.github/copilot-instructions.md

```diff
@@ -0,0 +1,154 @@
+# Darkflame Universe Server Development Instructions
+
+Darkflame Universe (DLU) is a LEGO Universe server emulator written in C++ with a multi-server architecture (AuthServer, ChatServer, MasterServer, WorldServer). Always reference these instructions first and fallback to search or bash commands only when you encounter unexpected information that does not match the info here.
+
+## Working Effectively
+
+### Bootstrap, Build, and Test - REQUIRED STEPS
+Execute these commands in order for ANY development work. NEVER CANCEL builds - they take time but work reliably:
+
+```bash
+# 1. Install system dependencies (Ubuntu/Debian)
+sudo apt update && sudo apt install -y build-essential gcc zlib1g-dev libssl-dev openssl mariadb-server cmake
+
+# 2. Initialize git submodules (CRITICAL - project won't build without this)
+git submodule update --init --recursive
+
+# 3. Build using the provided script
+./build.sh -j2
+```
+- **Build time: ~6 minutes. NEVER CANCEL. Set timeout to 720+ seconds (12+ minutes).**
+- **Uses CMake 3.25-3.31 (confirmed working with 3.31.6)**
+- **Requires g++11+ (confirmed working with g++ 13.3.0)**
+
+### Alternative Build Methods
+```bash
+# Using CMake presets (CI-style)
+cmake --workflow --preset ci-ubuntu-22.04
+
+# Manual CMake (for custom configurations)
+mkdir -p build && cd build
+cmake -DCMAKE_BUILD_TYPE="Release" ..
+cmake --build . --config Release -j2
+```
+- **Same timing: ~6 minutes. NEVER CANCEL. Set timeout to 720+ seconds.**
+
+### Run Tests
+```bash
+cd build
+ctest --output-on-failure
+```
+- **Test time: <4 seconds. Set timeout to 30+ seconds.**
+- **91 tests run, all should pass**
+- **Tests are built automatically when ENABLE_TESTING=1 in CMakeVariables.txt**
+
+### Database Setup (for runtime testing)
+```bash
+# Start MariaDB
+sudo systemctl start mysql
+
+# Create test database and user
+sudo mysql -e "CREATE USER 'testuser'@'localhost' IDENTIFIED BY 'testpass'; GRANT ALL ON *.* TO 'testuser'@'localhost' WITH GRANT OPTION; FLUSH PRIVILEGES; CREATE DATABASE testdarkflame;"
+```
+
+## Validation
+
+### Build Validation
+- **ALWAYS run the bootstrapping steps first** before making any code changes
+- **ALWAYS build and test your changes** before considering them complete
+- Build output should include all server binaries: AuthServer, ChatServer, MasterServer, WorldServer
+- Build directory contains required files: `*.ini` configs, `navmeshes/`, `migrations/`, `vanity/`, `blocklist.dcf`, `libmariadbcpp.so`
+
+### Runtime Validation
+The servers can be started for basic validation:
+```bash
+cd build
+./MasterServer
+```
+- **Server will start but complain about missing client files (this is expected)**
+- **Database connections work with proper configuration in sharedconfig.ini**
+- **For full server testing, LEGO Universe client files are required (not available in this repository)**
```

Comment:
What is the point of this without any client files?  no world servers can start and thats 90% of the codebase